src/mgr/DaemonState.h

```diff
@@ -355,6 +357,18 @@ class DaemonStateIndex
     return updating.count(k) > 0;
   }
 
+  void update_metadata(DaemonStatePtr state,
+		       const map<string,string>& meta) {
+    // remove and re-insert in case the device metadata changed
+    RWLock::WLocker l(lock);
+    _rm(state->key);
```

Comment:
Is the `_rm` redundant? It looks like `_insert` already invokes `_erase` before doing anything else.

Reply:
OK, got it.